share/spice/hearthstone/hearthstone.js

```diff
@@ -0,0 +1,53 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_hearthstone = function(api_result){
+
+        // Validate the response
+        if (!api_result || api_result.error || !api_result.Has_name || !api_result.page) {
+            return Spice.failed('hearthstone');
+        }
+        
+        // Card Data
+        // May (but very unlikely) return <undefined> values for : image_url, Has_card_type, Has_rarity,
+        // Has_class, Has_mana_cost
+        var card = {
+            page: api_result.page,
+            image: api_result.image_url,
+            name: api_result.Has_name,
+            type: api_result.Has_card_type,
+            rarity: api_result.Has_rarity,
+            hero: api_result.Has_class,
+            cost: api_result.Has_mana_cost,
+            attack: api_result.Has_attack, // <undefined> for cards other than minions
```

Comment:
If there's a chance of these being undefined, I would first check to see if it's defined, and if so, in your normalize function, create the property and set the value accordingly